### PR TITLE
Adding okta support for auth logins

### PR DESCRIPTION
**Pull request by Mike-Dunton**
This adds an "okta" option to the frontend and handles passing users credentials to `auth/okta/login/` 

Starting to see some code reuse in login.go. What are your thoughts? 

Also could you point to something explaining how the tests work? Are they integration tests? Are you mocking the auth backends?

### CODE REVIEW

File: frontend/client/views/login/index.vue [hunk 1]
```diff
@@ -112,6 +113,26 @@
                 </div>
               </div>
 
+              <!-- Userpass login form -->
```
Comment (Caiyeon) [inline frontend/client/views/login/index.vue]: Should be Okta instead of Userpass

**Comment by Caiyeon** [comment]
Yes, the backend login function should be just a simple refactor. I'll get to it in a couple of days unless you want to tackle it

The unit tests are definitely more integration than unit. A dev vault core is spun up in the backend before the tests, and provides a real vault to interact with. This vault dev core is version controlled via `vendor`, and should be locked to v0.7.3 at the moment.

There are no github or okta auth unit tests, simply because I haven't found the time to mess around with secrets (github in particular) on continuous integration. If you have an elegant way to do it, let me know

**Comment by Mike-Dunton** [comment]
I built this and deployed it to a staging vault cluster I maintain and was able to authenticate with the OKta backend. It returned a token and set the session in goldfish. The only thing It didn't do was do the notify since I did not have 'Okta' in the if line. I updated that.

**Comment by Caiyeon** [comment]
That's good enough for me. I'll merge this and you should see it in the 0.7.0 release

**Pull request merged**